prov/tcp/src/tcpx_conn_mgr.c

```diff
@@ -291,10 +192,16 @@ static int tcpx_ep_enable(struct tcpx_ep *ep,
 	ep->state = TCPX_CONNECTED;
 	ofi_mutex_unlock(&ep->lock);
 
-	ret = tcpx_ep_add_fd(ep);
+	ret = tcpx_watch_socket(domain, ep);
 	if (ret)
 		return ret;
 
+	// TODO: move to create/close - do we need state check in progress loop?
+	// want to continue to process receives as long as there's data
+	ofi_mutex_lock(&domain->util_domain.lock);
+	dlist_insert_tail(&ep->ep_entry, &domain->ep_list);
+	ofi_mutex_unlock(&domain->util_domain.lock);
```

Comment:
Shouldn't we move these calls in tcpx_watch_socket()?

Reply:
The TODO comment above the insertion is there to basically tell me that I don't like where this section of code is, but I need to see what happens when it is moved.  The current code separates keeping the ep on the ep_list from adding the socket to the pollfd, and there was a reason for that if I can just remember why.  But I don't know if that reason would still be valid.